### PR TITLE
Ftr: catch exceptions that user defined

### PR DESCRIPTION
**Pull request by cvictory**
1.当java抛出自定义业务异常的时候，当dubb-go client没有注册这个异常类型，这时候会出现错误。
这里会将这种异常统一转换为BizException。

2.原来的stackTraceElement数据丢失，进行修复。

之前的pr ： https://github.com/apache/dubbo-go-hessian2/pull/204  ，不小心在本地push错远程仓库，直接推进去了。 从代码层面进行了回滚，所以重新提交下PR。

### CODE REVIEW

File: java_unknown_exception.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	)
+	var count = 0
+	for _, item := range cls.fieldNameList {
+		if item == "detailMessage" || item == "suppressedExceptions" || item == "stackTrace" || item == "cause" {
+			count++
+		}
+	}
+	// if have these 4 fields, it is throwable struct
+	if count == 4 {
+		if throwable, ok = getStructInfo(cls.javaName); ok {
```
Comment (wongoo) [inline java_unknown_exception.go:48]: does it should move this check after the mutex.lock() ?

**Comment by wongoo** [comment]
@cvictory can u add a unit test that the exception data is from java, there are many ut samples in this project which decode bytes generated by java.

File: java_unknown_exception.go (hunk 1 above)
Comment (cvictory) [reply java_unknown_exception.go:50]: Done.

**Comment by cvictory** [comment]
> @cvictory can u add a unit test that the exception data is from java, there are many ut samples in this project which decode bytes generated by java.

增加了java的异常返回结果的用例，数据存在test_reource目录下的两个文件。

File: test_resource/RuntimeException.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Cjava.lang.RuntimeException�detailMessagecause
+stackTracesuppressedExceptions`test exceptQ�V[java.lang.StackTraceElement�?Cjava.lang.StackTraceElement�declaringClass
```
Comment (hxmhlt) [inline test_resource/RuntimeException.txt:2]: Why display not normal ?

**Comment by wongoo** [comment]
@cvictory  sorry for delay reply.  DO NOT add binary data into, which can't be checked, u can test java response easily, ref https://github.com/apache/dubbo-go-hessian2/blob/master/decode_test.go#L93

File: test_resource/RuntimeException.txt (hunk 2 above)
Comment (cvictory) [reply test_resource/RuntimeException.txt:2]: 存储在里面是以byte数组形式。

File: java_unknown_exception.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import (
+	"fmt"
+	"sync"
+)
+
+import (
+	"github.com/apache/dubbo-go-hessian2/java_exception"
+)
+
+var mutex sync.Mutex
```
Comment (zouyx) [inline java_unknown_exception.go:29]: I think should make this variable name more meaningful, like `getExceptionMutex`, `checkExceptionMutex` etc?

**Pull request merged**